examples/benchmark/benchmark-matmult.cpp

```diff
@@ -31,15 +32,58 @@ static void ggml_graph_compute_helper(std::vector<uint8_t> & buf, ggml_cgraph *
     ggml_graph_compute(graph, &plan);
 }
 
+#define QK8_0 32
+
+typedef struct {
+    uint16_t d;       // delta
+    int8_t qs[QK8_0]; // quants
+} block_q8_0;
+
 static float tensor_sum_elements(const ggml_tensor * tensor) {
-    double sum = 0;
+    double sum                  = 0;
+    float  floatvalue           = 0;
+    unsigned short shortvalue   = 0;
+
     if (tensor->type == GGML_TYPE_F32) {
         for (int j = 0; j < tensor->ne[1]; j++) {
             for (int k = 0; k < tensor->ne[0]; k++) {
-                sum += ((float *) tensor->data)[j*tensor->ne[0] + k];
+                sum += ((float *) tensor->data)[j * tensor->ne[0] + k];
             }
         }
     }
+
+    if (tensor->type == GGML_TYPE_I8) {
+        for (int j = 0; j < tensor->ne[1]; j++) {
+            for (int k = 0; k < tensor->ne[0]; k++) {
+                sum += ((int8_t *) tensor->data)[j * tensor->ne[0] + k];
+            }
+        }
+    }
+
+    if (tensor->type == GGML_TYPE_F16) {
+        for (int j = 0; j < tensor->ne[1]; j++) {
+            for (int k = 0; k < tensor->ne[0]; k++) {
+                shortvalue = ((unsigned short *) tensor->data)[j * tensor->ne[0] + k];
+                floatvalue = GGML_FP16_TO_FP32(shortvalue);
+                sum        += floatvalue;
+            }
+        }
+    }
+
+    if (tensor->type == GGML_TYPE_Q8_0) {
+        int blocks = 0;
+        block_q8_0 * quant_datas = (block_q8_0 *)tensor->data;
+        for (int j = 0; j < tensor->ne[1]; j++) {
+            blocks = tensor->ne[0] / QK8_0;
+            for (int i = 0; i < blocks; i++) {
+                floatvalue = GGML_FP16_TO_FP32(quant_datas[j * blocks + i].d);
```

Comment:
This is not correct for converting q8_0 to fp32. To convert any q type to fp32 (except for IQ), you can do:

```ggml_type_traits_t qtype = ggml_internal_get_type_traits(tensor->type);
qtype.to_float(data, f32_output, nelements);
```

Reply:
thanks your correction.I'll study and modify it accordingly.

Reply:
I'm not sure either, but it seems to me that the `dequantize_row_q8_0` calculates `const int nb = k / qk;` with k is the total number of elements, while in your implementation k = `tensor->ne[0]`

Reply:
k = tensor->ne[0] is used to calculate block counts.